src/Pages/Profile.js

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect, useContext } from 'react'
+import Card from '../components/UI/Card';
+import { AuthContext } from '../store/auth-context';
+
+const Profile = () => {
+  const [loading, setLoading] = useState(true);
+  const [username, setUsername] = useState(null);
+  const [website, setWebsite] = useState(null);
+  const [avatar_url, setAvatarUrl] = useState(null);
+  const [error, setError] = useState(null);
+
+  const authCtx = useContext(AuthContext);
+  const user = authCtx.user;
+
+  useEffect(() => {
+    (async () => {
+      try {
+        setLoading(true)
+        const { username, website, avatar_url } = await authCtx.getProfile();
+        setUsername(username);
+        setWebsite(website);
+        setAvatarUrl(avatar_url);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setLoading(false)
+      }
+    })();
```

Comment:
Osobně mám radši v těchto případech zápis bez obalování `async` funkcí:
```
setLoading(true);
authCtx.getProfile()
  .then(({ username, website, avatar_url }) => {
    setUsername(username);
    setWebsite(website);
    setAvatarUrl(avatar_url);
  })
  .catch((error) => setError(error))
  .finally(() => setLoading(false));
```
Ale tohle je v pohodě a lidi u mě na projektu to (přes moje protesty) dělají :smile: